pkg/client/typed/dynamic/client.go

```diff
@@ -49,10 +49,6 @@ func NewClient(conf *restclient.Config) (*Client, error) {
 
 	conf.Codec = dynamicCodec{}
 
-	if conf.APIPath == "" {
```

Comment:
This seems unrelated to the rest of this pull. How is this PR using the dynamic client differently such that this was needed?